src/compiler/checker.ts

```diff
@@ -18784,6 +18790,34 @@ namespace ts {
             return suggestion && symbolName(suggestion);
         }
 
+        function getSuggestionForNonexistentIndexSignature(objectType: Type, expr: ElementAccessExpression): string | undefined {
+            let suggestion: string | undefined;
+            // check if object type has setter or getter
+            const hasProp = (name: "set" | "get") => {
+                const prop = getPropertyOfObjectType(objectType, <__String>name);
+                if (prop) {
+                    const s = getSingleCallSignature(getTypeOfSymbol(prop));
+                    if (s && getMinArgumentCount(s) === 1 && typeToString(getTypeAtPosition(s, 0)) === "string") {
+                        return true;
+                    }
+                }
+                return false;
+            };
+
+            if (isAssignmentTarget(expr)) {
+                if (hasProp("set")) {
+                    suggestion = symbolToString(objectType.symbol) + ".set";

```

Comment:
Unfortunately `symbolToString` might not work if the expression isn't a simple binding. You'd be better off with something like

```ts
const suggestedMethod = isAssignmentTarget(expr) ? "set" : "get";
if (!hasProp(suggestedMethod)) {
    return undefined;
}

let suggestion = tryGetPropertyAccessOrIdentifierToString(expr);
if (suggestion === undefined) {
    suggestion = suggestedMethod;
}
else {
    suggestion += "." + suggestedMethod;
}
```

where you could define `tryGetPropertyAccessOrIdentifierToString` in `utilities.ts`.

```ts
function tryGetPropertyAccessOrIdentifierToString(expr: Expression): string | undefined {
    if (isPropertyAccessExpression(expr)) {
        return tryGetPropertyAccessOrIdentifierToString(expr.expression) + "." + x.name;
    }
    if (isIdentifier(expr)) {
        return unescapeLeadingUnderscores(expr.escapedText);
    }
    return undefined;
}
```

Reply:
Right! Let me do this in a few. Thank you.